### PR TITLE
Adding edition picker to the new nav

### PR DESCRIPTION
**Pull request by NataliaLKB**
## What does this change?

Making the edition picker in the new nav work. It is important that this works without JavaScript, as this will also be available to core users and no js users.

My [first attempt](https://github.com/guardian/frontend/pull/13617) at this was using the `<select>` and `<option>` elements. However I ran into some [accessibility concerns with IE](https://github.com/guardian/frontend/pull/13617#discussion_r70998183). 
## What is the value of this and can you measure success?

I am aiming to run an a/b test for this in the next couple weeks!
## Does this affect other platforms - Amp, Apps, etc?

Nope
## Screenshots

editionpickerstyled

### CODE REVIEW

File: static/src/stylesheets/layout/_ab-new-header-test-variant.scss [hunk 1]
```diff
@@ -241,6 +204,93 @@ $gutter-medium: 37px;
...
+
+    &:hover {
+        fill: #ffffff;
+    }
+}
+
+.edition-picker__checkbox {
+    display: none;
+
+    &:checked {
```
Comment (regiskuckaertz) [inline static/src/stylesheets/layout/_ab-new-header-test-variant.scss]: This is smart! To make this link explicit, you can also add the `aria-controls` attribute and then add/remove the `hidden` attribute on the list.

Comment (NataliaLKB) [inline static/src/stylesheets/layout/_ab-new-header-test-variant.scss]: Yes that makes sense! I will do that

Comment (NataliaLKB) [inline static/src/stylesheets/layout/_ab-new-header-test-variant.scss]: Changes we discussed have been made. Thanks for those suggestions @regiskuckaertz

File: static/src/javascripts/projects/common/modules/navigation/edition-picker.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/$'
+], function (
+    fastdomPromise,
+    $
+) {
+    var dropdown = $('.js-edition-picker-dropdown');
+
+    function enhanceToButton() {
```
Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/navigation/edition-picker.js]: Well, this is just beautiful, my only bother when reading the mark-up that it wasn't a button, but lo-and-behold you have it covered ✊

File: static/src/javascripts/projects/common/modules/navigation/edition-picker.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    function openMenuOnClick() {
+        var button = $('.js-click');
+
+        if (button.length > 0) {
+            button[0].addEventListener('click', function() {
+
+                fastdomPromise.write(function() {
+                    if (button.hasClass('open')) {
+                        button.removeClass('open');
+                        dropdown.attr('hidden');
```
Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/navigation/edition-picker.js]: Does this set it as expected? I thought a single value in `attr` was a getter only but maybe it does something special with boolean attributes

Comment (NataliaLKB) [inline static/src/javascripts/projects/common/modules/navigation/edition-picker.js]: Fixed @gtrufitt 😄

File: static/src/javascripts/projects/common/modules/navigation/edition-picker.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        fastdomPromise.write(function() {
+            checkBox.remove();
+            container.prepend(button);
+
+            openMenuOnClick();
+        });
+    }
+
+    function openMenuOnClick() {
+        var button = $('.js-click');
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/navigation/edition-picker.js]: Maybe this class could be more descriptive, or more certain to be unique to the `edition-picker`?

Comment (NataliaLKB) [inline static/src/javascripts/projects/common/modules/navigation/edition-picker.js]: good shout! I will change it 😄

File: static/src/javascripts/projects/common/modules/navigation/edition-picker.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            });
+        }
+    }
+
+    function clickMenuOnEnter() {
+        var label = document.getElementsByClassName('js-on-enter-click')[0];
+
+        if (label) {
+            label.onkeypress =  function clickOnEnter(event) {
+                if((event.keyCode ? event.keyCode : event.which) == 13) {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/navigation/edition-picker.js]: Triple eqs all the way! `===`

File: static/src/stylesheets/layout/_ab-new-header-test-variant.scss [hunk 6]
```diff
@@ -8,8 +8,9 @@ $mobile-medium: 375px; // Breakpoint for our most common device sizes
 }
 
 .new-header__logo-wrapper {
-    padding-top: $gs-baseline / 2;
-    padding-bottom: $gs-baseline / 2;
+    padding-top: 6px;
```
Comment (SiAdcock) [inline static/src/stylesheets/layout/_ab-new-header-test-variant.scss]: I think `gs-baseline` is `12px`, so we can stick with `$gs-baseline / 2`

File: static/src/javascripts/projects/common/modules/navigation/edition-picker.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+                        button.attr('aria-expanded', 'true');
+                        dropdown.removeAttr('hidden');
+                    }
+                });
+            });
+        }
+    }
+
+    function clickMenuOnEnter() {
+        var label = document.getElementsByClassName('js-on-enter-click')[0];
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/navigation/edition-picker.js]: Is it possible for an element with this class to *not* be present on the page? If so, this line will fall over (whereas the pattern used on lines 29 - 32 above is more defensive)

Comment (NataliaLKB) [inline static/src/javascripts/projects/common/modules/navigation/edition-picker.js]: It is always possible! I made some amendments. Thanks 😄

**Comment by SiAdcock** [comment]
Very cool stuff 👍

**Pull request merged**